hack/cpaas/patch-cluster-with-rhpm-0.1.sh

```diff
@@ -1,96 +1,108 @@
-    if [[ $version == 0.1* ]] && [[ $ocp_version == "$requested_ocp_version" ]];
-    then
-      index_image=$(echo $message | jq -r '.msg.index.index_image')
-      break
-    else
-      i=$(expr $i + 1)
-    fi
-  done
-  ok "index image found: $index_image"
-  brew_index_image=$(echo $index_image | awk -F':' '{print "brew.registry.redhat.io/rh-osbs/iib:"$2}')
-  echo $brew_index_image
+validate_podman() {
+	header "Validating podman"
+
+	command -v podman >/dev/null 2>&1 || {
+		fail "No podman found"
+		info "Please install podman or make sure its running"
+		return 1
+	}
 }
 
-function add_brew_registry_credentials(){
-  header "Getting credentials for brew registry"
-  info "getting token from employee-token-manager"
-  local tokens=$(curl --negotiate -u : https://employee-token-manager.registry.redhat.com/v1/tokens -s)
-  if [[ "$?" != "0" ]] || [[ "$tokens" == "null" ]] || [[ "$tokens" == "" ]];
-  then
-	err "could not get token. \n\
-    please use the following command to create a token and retry the script. \n\
-    curl --negotiate -u : -X POST -H 'Content-Type: application/json' --data '{\"description\":\"for testing cpaas built powermon images on openshift 4 cluster\"}' https://employee-token-manager.registry.redhat.com/v1/tokens -s
-	"
-	exit -1
-  fi
-  ok "found token..."
-  local username=$(echo $tokens | jq 'last' | jq -r  '.credentials.username')
-  local password=$(echo $tokens | jq 'last' | jq -r  '.credentials.password')
-  #info "username: $username, password: $password"
-
-  info "getting auth from cluster"
-  run oc get secret/pull-secret -n openshift-config -o json | jq -r '.data.".dockerconfigjson"' | base64 -d > authfile
-
-  info "logging to brew registry"
-  run podman login --authfile authfile --username "$username" --password "$password" brew.registry.redhat.io
-
-  info "set auth to cluster"
-  run oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=authfile
+get_index_image() {
+	header "Fetch index image"
+	local url="https://datagrepper.engineering.redhat.com/raw?topic=/topic/VirtualTopic.eng.ci.redhat-container-image.index.built&delta=824000&contains=$POWERMON_BUNDLE"
+	local ret=0
+
+	INDEX_IMG=$(curl -s "$url" |
+		jq --arg requested_ocp_version "$OCP_VERSION" -r \
+			'.raw_messages[] |
+      select(.msg.index.ocp_version == $requested_ocp_version) |
+        .msg.index.index_image' |
+		head -n 1 | awk -F':' '{print "brew.registry.redhat.io/rh-osbs/iib:"$2}')
+
+	[[ -n $INDEX_IMG && $INDEX_IMG != "null" ]] || {
+		ret=1
+		err "No matching index image found. Please check if provided OCP version is available or connected to VPN!"
+		return $ret
+	}
+	ok "Using index image: $INDEX_IMG"
+	return $ret
+
 }
 
+add_brew_registry() {
+	header "Getting credentials for brew registry"
+	local url="https://employee-token-manager.registry.redhat.com/v1/tokens"
+	local ret=0
+
+	! token=$(curl --negotiate -u : $url -s) || [[ -z $token ]] || [[ $token == "null" ]] && {
```

Comment:
IMHO, this is overly terse for really no good reason.

Reply:
Agree. I didn't want to play with the original script too much